gpustack/routes/auth.py

```diff
@@ -429,8 +430,30 @@ async def login(
 
 
 @router.post("/logout")
-async def logout(response: Response):
-    response.delete_cookie(key=SESSION_COOKIE_NAME)
+async def logout(request: Request):
+    config: Config = request.app.state.server_config
+    oidc_redirect_url = None
+    if (
+        config.external_auth_type == AuthProviderEnum.OIDC
+        and config.openid_configuration
+    ):
+        end_session_endpoint = config.openid_configuration.get("end_session_endpoint")
+        if end_session_endpoint:
+            redirect_uri = str(config.server_external_url or request.base_url)
+            params = {
+                "client_id": config.oidc_client_id,
+                "post_logout_redirect_uri": redirect_uri,
+            }
```

Comment:
We may need to include `id_token_hint`. It's RECOMMENDED in https://openid.net/specs/openid-connect-rpinitiated-1_0.html